src/Models/MenuItem.php

```diff
@@ -66,6 +66,24 @@ public function menu()
         return $this->belongsTo(Menu::class, 'menu_id');
     }
 
+    public function url(): Attribute
+    {
+        return new Attribute(function ($value) {
+            switch($this->type) {
+                case 'index-page':
+                    $resourceClass = $this->reference_page;
+                    return route($resourceClass::getFrontendIndexRoute());
+                case 'content':
+                    $resourceClass = $this->reference_page;
```

Comment:
This could be put before the switch so that it does not need to defined multiple times